tests/unit/amo/test_withExperiment.js

```diff
@@ -91,6 +96,43 @@ describe(__filename, () => {
     expect(root).toHaveProp('variant', variantId);
   });
 
+  it('injects a variant prop from the redux store', () => {
+    const { store } = dispatchClientMetadata();
+    const id = makeId('test-id');
+    const variantId = 'some-variant-id';
+    const cookies = fakeCookies({
+      get: sinon.stub().returns(undefined),
+    });
+
+    store.dispatch(
+      storeExperimentVariant({ storedVariant: { id, variant: variantId } }),
+    );
+
+    const root = render({ cookies, experimentProps: { id }, store });
+
+    expect(root).toHaveProp('variant', variantId);
+  });
+
+  it('prefers a variant from a cookie to one from the the redux store', () => {
```

Comment:
it'd be interesting to explain why.

Reply:
Sure, I can do that. I didn't want to clutter the test with comments but I see your point. The reasoning is that if a cookie has been set, we should always respect that, because our goal is consistency for a user throughout the experiment.